lib/scala/logging-config/src/main/java/org/enso/logger/config/LoggersLevels.java

```diff
@@ -0,0 +1,97 @@
+package org.enso.logger.config;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.event.Level;
+
+/** Encapsulates custom log levels that can be set via config file and environmental variables. */
+public class LoggersLevels {
+
+  private Map<String, Level> loggers;
+
+  private LoggersLevels(Map<String, Level> loggers) {
+    this.loggers = loggers;
+  }
+
+  public Set<Map.Entry<String, Level>> entrySet() {
+    return loggers.entrySet();
+  }
+
+  public static LoggersLevels parse() {
+    return parse(ConfigFactory.empty());
+  }
+
+  public static LoggersLevels parse(Config config) {
+    // LinkedHashMap ensures that wildcard loggers are de-prioritized
+    Map<String, Level> loggers = systemLoggers();
+    Map<String, Level> fallbacks = new LinkedHashMap<>();
+    config
+        .entrySet()
+        .forEach(
```

Comment:
You can call [`Map.forEach`](https://javadoc.scijava.org/Java17/java.base/java/util/Map.html#forEach(java.util.function.BiConsumer)) directly

Reply:
Nope. `config` is not a map.